File: backend/src/main/java/com/gxs/bank/controller/PromotionController.java
package com.gxs.bank.controller;

import com.gxs.bank.dto.response.ApiResponse;
import com.gxs.bank.model.Promotion;
import com.gxs.bank.service.PromotionService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/promotions")
@RequiredArgsConstructor
public class PromotionController {

    private final PromotionService promotionService;

    @GetMapping
    public ResponseEntity<ApiResponse<List<Promotion>>> getActivePromotions() {
        List<Promotion> promotions = promotionService.getActivePromotions();
        return ResponseEntity.ok(ApiResponse.ok(promotions));
    }

    @GetMapping("/{id}")
    public ResponseEntity<ApiResponse<Promotion>> getPromotion(@PathVariable UUID id) {
        Promotion promotion = promotionService.getPromotion(id);
        return ResponseEntity.ok(ApiResponse.ok(promotion));
    }
}

File: backend-python/com/gxs/bank/controller/PromotionController.py
from fastapi import APIRouter, Depends

from com.gxs.bank.config.database import get_db
from com.gxs.bank.config.serialization import serialize
from com.gxs.bank.dto.response.ApiResponse import ok
from com.gxs.bank.service.PromotionService import PromotionService


router = APIRouter(prefix="/api/promotions", tags=["Promotion"])


@router.get("")
def getActivePromotions(db=Depends(get_db)):
    promotions = PromotionService(db).getActivePromotions()
    return ok(serialize(promotions))


@router.get("/{promotion_id}")
def getPromotion(promotion_id: str, db=Depends(get_db)):
    promotion = PromotionService(db).getPromotion(promotion_id)
    return ok(serialize(promotion))
